### PR TITLE
Feature/issue6 malformed message

### PR DESCRIPTION
**Pull request by sharonx**
- Here malformed message only happens when an event message is an empty string.
- The other malformed situation is non utf8 encoded bytes. 
 - Added more debug messages

### CODE REVIEW

**Comment by dtregonning** [comment]
Thanks for PR @sharonx 

- All the log additions are good.
- Is checking every log message and splicing into a byte[][] the most efficient way to eliminate the event of only spaces issues? Would it better to trigger the check for the malformed message while JSON marshaling and handling there?

**Comment by sharonx** [comment]
@dtregonning HEC doesn't accept empty string. But empty string is acceptable by JSON as value. 

Here I'm trying to decode the message while reading from the fifo. If it is empty, the message is discarded so it won't make it to the json marshal stage.

**Review by dtregonning — APPROVED** [review]
Thanks for clearing up empty line.

File: hec_client.go [hunk 1]
```diff
@@ -31,12 +31,14 @@ type hecClient struct {
...
 
 func (hec *hecClient) postMessages(messages []*splunkMessage, lastChance bool) []*splunkMessage {
+	logrus.Debugf("Received %d messages.", len(messages))
 	messagesLen := len(messages)
 	for i := 0; i < messagesLen; i += hec.postMessagesBatchSize {
 		upperBound := i + hec.postMessagesBatchSize
 		if upperBound > messagesLen {
 			upperBound = messagesLen
 		}
+		logrus.WithField("upperBound", upperBound).WithField("messagesLen", messagesLen).Debug("processing batch")
```
Comment (chenziliang) [inline hec_client.go]: Would avoid doing logging in a loop even it is debug

File: hec_client.go [hunk 2]
```diff
@@ -64,7 +66,9 @@ func (hec *hecClient) postMessages(messages []*splunkMessage, lastChance bool) [
 }
 
 func (hec *hecClient) tryPostMessages(messages []*splunkMessage) error {
+	logrus.Debugf("try to post %d messages.", len(messages))
```
Comment (chenziliang) [inline hec_client.go]: Same as above

File: hec_client.go [hunk 3]
```diff
@@ -84,6 +88,7 @@ func (hec *hecClient) tryPostMessages(messages []*splunkMessage) error {
 	}
 	for _, message := range messages {
 		jsonEvent, err := json.Marshal(message)
+		logrus.Debugf("Message event is %s", jsonEvent)
```
Comment (chenziliang) [inline hec_client.go]: Same as above

File: splunk_logger.go [hunk 4]
```diff
@@ -497,6 +495,9 @@ func (l *splunkLogger) worker() {
 	for {
 		select {
 		case message, open := <-l.stream:
+			if message != nil {
+				logrus.Debugf("Reading a message event %s", message.Event)
```
Comment (chenziliang) [inline splunk_logger.go]: Same as above

**Comment by sharonx** [comment]
@chenziliang thanks for the comments. I updated the PR.

**Pull request merged**